WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/SetUpTapToPayViewModelsOrderedList.swift

```diff
@@ -0,0 +1,70 @@
+import Foundation
+import Yosemite
+
+/// Aggregates an ordered list of viewmodels, conforming to the viewmodel provider protocol. Priority is given to
+/// the first viewmodel in the list to return true for shouldShow
+///
+final class SetUpTapToPayViewModelsOrderedList: PaymentSettingsFlowPrioritizedViewModelsProvider {
+
+    private var viewModelsAndViews = [PaymentSettingsFlowViewModelAndView]()
+
+    var priorityViewModelAndView: PaymentSettingsFlowViewModelAndView? {
+        didSet {
+            onPriorityChanged?(priorityViewModelAndView)
+        }
+    }
+
+    var onPriorityChanged: ((PaymentSettingsFlowViewModelAndView?) -> ())?
+
+    private var knownReaderProvider: CardReaderSettingsKnownReaderProvider?
+
+    private let cardReaderConnectionAnalyticsTracker: CardReaderConnectionAnalyticsTracker
+
+    init(siteID: Int64,
+         configuration: CardPresentPaymentsConfiguration,
+         activePaymentGateway: CardPresentPaymentsPlugin) {
+        /// Initialize dependencies for viewmodels first, then viewmodels
+        ///
+        cardReaderConnectionAnalyticsTracker = CardReaderConnectionAnalyticsTracker(configuration: configuration)
+
+        /// Instantiate and add each viewmodel related to setting up Tap to Pay on iPhone to the
+        /// array. Viewmodels will be evaluated for shouldShow starting at the top
+        /// of the array. The first viewmodel to return true for shouldShow is given
+        /// priority, so viewmodels related to starting set up should come before viewmodels
+        /// that expect set up to be completed, etc.
+        ///
+        viewModelsAndViews.append(
+            PaymentSettingsFlowViewModelAndView(
+                viewModel: SetUpTapToPayInformationViewModel(
+                    siteID: siteID,
+                    configuration: configuration,
+                    didChangeShouldShow: { [weak self] state in
+                        self?.onDidChangeShouldShow(state)
+                    },
+                    activePaymentGateway: activePaymentGateway,
+                    connectionAnalyticsTracker: cardReaderConnectionAnalyticsTracker
+                ),
+                viewPresenter: SetUpTapToPayInformationViewController.self
+            )
+        )
```

Comment:
I understand Unit Tests will come later outside of this PR, so this might be already taken into account: Should we inject these into the initializer, and extract the logic for `viewModelsAndViews.append` into a different method?

Reply:
Yeah it probably should be in its own method, I'll have to look at that in future. 

I think of this init method more like a definition for the flow, so we won't start injecting the `viewModelsAndViews` that we want to have here, but it could be neater.